### DISCUSSION TITLE
Inquiry about underlying technologies in Aries/Credo framework

### DISCUSSION THREAD

**Discussion by davideroffo** [openwallet-foundation/credo-ts#1890] [Q&A] [answered]
Dear all,

I am currently working on a project that utilises the Aries/Credo framework. To gain a more comprehensive understanding of its inner workings, I'm reaching out to inquire about the technical foundation upon which the Aries/Credo-TS libraries are built.

Specifically, I am interested in:

- The programming languages employed in the development of the Aries/Credo-TS libraries. Are they primarily written in JavaScript or TypeScript, or do they leverage other underlying technologies?

Any information pertaining to the underlying technologies that underpin the Aries/Credo-TS services would be immensely valuable. This knowledge will empower me to approach development tasks associated with the project in a more efficient and effective manner.

Thank you for your time and assistance.

Sincerely,
Davide

**Comment by TimoGlastra** [comment, marked as answer]
Hey Davide, 

Credo is 100% written in typescript, but it does rely on dependencies written in other languages:
- AnonCreds RS, Indy VDR, Aries Askar are written in Rust and called as a c-callable binary from Node.JS or React Native
  - The react native wrappers for this are written in C++
- We will add support for mDL credentials in the near future which will be based on a kotlin multi-platform library

Usually when you want to work with or extend the framework it's not needed to work directly with the underlying dependencies and thus TypeScript is the main language involved.

does this answer your question?
